helpers/compile/plugins/fill-plugin/fillers/buffer-small.ts

```diff
@@ -0,0 +1,504 @@
+  writeUint8(value: number, offset: number): void
+  writeUInt8(value: number, offset: number): void
+  writeUintBE(value: number, offset: number, byteLength: number): void
+  writeUIntBE(value: number, offset: number, byteLength: number): void
+  writeUintLE(value: number, offset: number, byteLength: number): void
+  writeUIntLE(value: number, offset: number, byteLength: number): void
+}
+
+export class Buffer extends Uint8Array /* implements NodeBuffer */ {
+  readonly _isBuffer = true
+
+  get offset() {
+    return this.byteOffset
+  }
+
+  static alloc(size: number, fill: string | number | Uint8Array = 0, encoding?: Encoding) {
+    return Buffer.allocUnsafe(size).fill(fill, encoding)
+  }
+
+  static allocUnsafe(size: number) {
+    return new Buffer(new Uint8Array(size))
+  }
+
+  static allocUnsafeSlow(size: number) {
+    return new Buffer(new Uint8Array(size))
+  }
+
+  static isBuffer(arg: any): arg is Buffer {
+    return arg && !!arg._isBuffer
+  }
+
+  static byteLength(string: string, encoding: Encoding = 'utf8') {
+    return Buffer.from(string, encoding).length
+  }
+
+  static compare(a: Uint8Array, b: Uint8Array) {
+    for (let i = 0; i < a.length; i++) {
+      if (a[i] < b[i]) return -1
+      if (a[i] > b[i]) return 1
+    }
+
+    return a.length === b.length ? 0 : a.length > b.length ? 1 : -1
+  }
+
+  static from(value: unknown, encoding: any = 'utf8'): Buffer {
+    if (value && typeof value === 'object' && value['type'] === 'Buffer') {
+      return new Buffer(value['data'])
+    }
+
+    if (typeof value === 'number') {
+      return new Buffer(new Uint8Array(value))
+    }
+
+    if (typeof value === 'string' && typeof encoding === 'string') {
+      return stringToBuffer(value, encoding)
+    }
+
+    if (typeof value === 'object' && ArrayBuffer.isView(value)) {
+      return new Buffer(value.buffer, value.byteOffset, value.byteLength)
+    }
+
+    return new Buffer(value as ArrayLike<number>)
+  }
+
+  static concat(list: readonly Uint8Array[], totalLength?: number) {
+    if (list.length === 0) return Buffer.alloc(0)
+
+    const concat = ([] as number[]).concat(...list.map((item) => [...item]))
+    const result = Buffer.alloc(totalLength !== undefined ? totalLength : concat.length)
+
+    result.set(totalLength !== undefined ? concat.slice(0, totalLength) : concat)
+
+    return result
+  }
+
+  slice(start = 0, end = this.length) {
+    return this.subarray(start, end)
+  }
+
+  subarray(start = 0, end = this.length) {
+    return Object.setPrototypeOf(super.subarray(start, end), Buffer.prototype) as Buffer
+  }
+
+  reverse() {
+    super.reverse()
+
+    return this
+  }
+
+  readIntBE(offset: number, byteLength: number) {
+    const view = new DataView(this.buffer, offset, byteLength)
+    let val = 0
+    for (let i = 0; i < byteLength; i++) {
+      val = val * 0x100 + view.getUint8(i)
+    }
+
+    if (view.getUint8(0) & 0x80) {
+      val -= Math.pow(0x100, byteLength)
+    }
+
+    return val
+  }
```

Comment:
Not written by me, just taken from the Node.js 0.x repository and upgraded with a `DataView`.